spring-ai-client-chat/src/main/java/org/springframework/ai/chat/client/advisor/PromptChatMemoryAdvisor.java

```diff
@@ -149,30 +161,119 @@ private void after(ChatClientResponse chatClientResponse) {
+		}
+
+		if (!assistantMessages.isEmpty()) {
+			this.chatMemory.add(this.doGetConversationId(chatClientResponse.context()), assistantMessages);
+			logger.debug("[PromptChatMemoryAdvisor.after] Added ASSISTANT messages to memory for conversationId={}: {}",
+					this.doGetConversationId(chatClientResponse.context()), assistantMessages);
+			List<Message> memoryMessages = this.chatMemory.get(this.doGetConversationId(chatClientResponse.context()));
+			logger.debug("[PromptChatMemoryAdvisor.after] Memory after ASSISTANT add for conversationId={}: {}",
+					this.doGetConversationId(chatClientResponse.context()), memoryMessages);
+		}
+		return chatClientResponse;
+	}
+
+	@Override
+	public Flux<ChatClientResponse> adviseStream(ChatClientRequest chatClientRequest,
+			StreamAdvisorChain streamAdvisorChain) {
+		// Get the scheduler from BaseAdvisor
+		Scheduler scheduler = this.getScheduler();
+
+		// Process the request with the before method
+		return Mono.just(chatClientRequest)
+			.publishOn(scheduler)
+			.map(request -> this.before(request, streamAdvisorChain))
+			.flatMapMany(streamAdvisorChain::nextStream)
+			.transform(flux -> new MessageAggregator().aggregateChatClientResponse(flux,
+					response -> this.after(response, streamAdvisorChain)));
 	}
 
-	public static class Builder extends AbstractChatMemoryAdvisor.AbstractBuilder<ChatMemory> {
+	/**
+	 * Builder for PromptChatMemoryAdvisor.
+	 */
+	public static class Builder {
 
 		private PromptTemplate systemPromptTemplate = DEFAULT_SYSTEM_PROMPT_TEMPLATE;
 
+		private String conversationId = ChatMemory.DEFAULT_CONVERSATION_ID;
+
+		private int order = Advisor.DEFAULT_CHAT_MEMORY_PRECEDENCE_ORDER;
+
+		private Scheduler scheduler = BaseAdvisor.DEFAULT_SCHEDULER;
+
+		private ChatMemory chatMemory;
+
 		protected Builder(ChatMemory chatMemory) {
-			super(chatMemory);
+			this.chatMemory = chatMemory;
 		}
 
+		/**
+		 * Set the system text advice.
+		 * @param systemTextAdvise the system text advice
+		 * @return the builder
+		 */
 		public Builder systemTextAdvise(String systemTextAdvise) {
 			this.systemPromptTemplate = new PromptTemplate(systemTextAdvise);
 			return this;
 		}
 
+		/**
+		 * Set the system prompt template.
+		 * @param systemPromptTemplate the system prompt template
+		 * @return the builder
+		 */
 		public Builder systemPromptTemplate(PromptTemplate systemPromptTemplate) {
 			this.systemPromptTemplate = systemPromptTemplate;
 			return this;
 		}
 
+		/**
+		 * Set the conversation id.
+		 * @param conversationId the conversation id
+		 * @return the builder
+		 */
+		public Builder conversationId(String conversationId) {
+			this.conversationId = conversationId;
+			return this;
+		}
+
+		/**
+		 * Set whether to protect from blocking.
+		 * @param protectFromBlocking whether to protect from blocking
+		 * @return the builder
+		 */
+		public Builder protectFromBlocking(boolean protectFromBlocking) {
```

Comment:
I would also remove this in favour of `scheduler()`.

Reply:
We have this in `QuestionAnswerAdvisor` as well so leaving for consistency unless we want to remove from there (and others).  Seems like a nice helper.